### DISCUSSION TITLE
Custom plugin - how to display header, footer and possibly sidebar?

### DISCUSSION THREAD

**Discussion by ahtremblay** [facebook/docusaurus#7128] [Q&A] [answered]
I have a file `login.md` with content:

`# login`

I am using this code as a "pseudo-example": [docusaurus-plugin-content-pages](https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-plugin-content-pages/src/index.ts)

The code of my custom plugin is as follows

```
module.exports = function (context, options) {

    return {
        name: 'docusaurus-plugin-accounts',
        async contentLoaded({content, actions}) {
            const {addRoute} = actions;

            addRoute({
                path: '/account',
                component: '@site/plugins/accounts/login.md',
                exact: true,
            });
        },
    }
}
```

When I navigate to /account I get a page what shows the word login, as per the markdown. But there are no headers or footers - only login. How do I make it so I get the header and the footer to be display on the pages routed by my custom plugin? Here I am talking about the header and the footer that are defined in the docusaurus.config.js file.

**Comment by slorber** [comment, marked as answer]
By default a Docusaurus route doesn't have any layout.

If you use a markdown file as a component, then it doesn't have a layout either by default: you should apply the layout on top of this component.


This works:

```js
function myPlugin() {
      return {
        name: 'docusaurus-plugin-accounts',
        async contentLoaded({actions}) {
          const {addRoute} = actions;
          addRoute({
            path: '/account',
            component: '@site/src/accounts/AccountPage.js',
            modules: {
              AccountContentComponent: '@site/src/accounts/login.md',
            },
            exact: true,
          });
        },
      };
    },
```

```js
import React from 'react';
import Layout from '@theme/Layout';

export default function AccountPage({AccountContentComponent}) {
  return (
    <Layout>
      <h1>Account Page</h1>
      <AccountContentComponent />
    </Layout>
  );
}
```

```md
## Account Login

Sign in to your account.
```

<img width="1916" alt="image" src="https://user-images.githubusercontent.com/749374/162424983-2c27baae-85f7-4d4a-a8ac-c0c9337b9010.png">

**Comment by ahtremblay** [comment]
Thanks for the answer.

How do I get this dependency?

`import Layout from '@theme/Layout';`

All I have that contains layout in my project is:

`import Layout from "@docusaurus/core/lib/client/theme-fallback/Layout";`

**Reply by ahtremblay** [reply]
Well I created a new project and I got `'@theme/Layout'` in it. Not sure what happened the first time around and why I didn't have it. I will mark the question as answered. Thanks.
